### ISSUE TITLE
LTS 2.28 license

### ISSUE THREAD

**Issue by Tachi107** [ARMmbed/mbedtls#5408]
Hi, sorry if I'm asking this here and not on the mailing list but it seems that its [webpage is broken](https://lists.trustedfirmware.org/mailman/listinfo/mbed-tls), and I don't know what the address is.

In the Readme it is stated that contributions have to be licensed under both the Apache-2.0 and the GPL-2.0-or-later, because LTS branches are licensed under either the first or the latter. This was also reflected in the previous LTS branch, 2.16, as in its maintenance branch all the license headers in the source files were updated to reflect the dual licensing (see https://github.com/ARMmbed/mbedtls/commit/f744bd72eec35005e5d3a4bf1192014d15bef84f).

License headers in 2.28 have not been updated, and I don't know if this was intended or if this was a simple oversight. If LTS branches are now licensed only under the Apache-2.0 it is a big change, expecially for distributions that have strong licensing rules (e.g. Debian), and it would be nice to have an official statement on this matter.

Thanks :)

**Comment by yanesca** [comment]
Hi @Tachi107 , thank you for letting us know that the mailing list link is broken. The software has been updated and the links have changed, we'll fix the links soon: https://github.com/ARMmbed/mbedtls/pull/5420

Regarding the license, having Apache-2.0 only in 2.28 is intentional. 2.16 had GPL-2.0, because it started its life as an LTS with GPL-2.0 still present. By the time 2.28 became an LTS, GPL-2.0 was dropped a long time ago.

**Comment by yanesca** [comment]
Sorry for the misclick, reopening.

**Comment by Tachi107** [comment]
Ok, thanks for the clarification :)

Contributors now only need to accept the Apache-2.0, right? If so, I would suggest updating the Readme

**Comment by Tachi107** [comment]
Out of curiosity, why was the GPL-2.0 for LTS branches completely dropped? And more importantly, can projects licensed under the GPL-2.0 continue to use MbedTLS?

**Comment by daverodgman** [comment]
As Janos says, we stopped releasing under the GPLv2 in early 2020, and now release only under Apache 2.0 (all releases since 2.16 have been Apache 2.0 licensed). The reference to accepting submissions under GPL enabled us to continue supporting our existing 2.16 release, but is not relevant for any later branches. (We will likely look at updating the readme at some point).

We found that a number of people interested in the project preferred this licensing arrangement, so whilst unfortunately this does make the project unsuitable for users who require it under a GPLv2 license, it is beneficial for users who prefer the current arrangement.

Given that Mbed TLS is now released under only an Apache 2.0 license, we advise users to obtain their own legal advice to  confirm whether it is possible to include Mbed TLS in projects released under other licenses.
